test/rpc/test_data/get_events.py

```diff
@@ -8,120 +8,65 @@
     PREDEPLOY_ACCOUNT_CLI_ARGS,
     PREDEPLOYED_ACCOUNT_ADDRESS,
 )
+from typing import List, Optional, Union
 
 from starknet_devnet.blueprints.rpc.utils import rpc_felt
 
-BLOCK_FROM_0_TO_LATEST = {
-    "from_block": {"block_number": 0},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-        rpc_felt(INCREASE_BALANCE_CALLED_EVENT_KEY),
-    ],
-    "to_block": "latest",
-    "chunk_size": 10,
-}
 
-BLOCK_FROM_0x0_TO_LATEST = {
-    "from_block": {"block_hash": rpc_felt("0x0")},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "to_block": "latest",
-    "chunk_size": 10,
-}
-
-BLOCK_FROM_0_TO_LATEST_MALFORMED_REQUEST = {
-    "from_block": {"block_number": 0},
-    "to_block": "latest",
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "chunk_size": "test",
-}
-
-BLOCK_FROM_0_TO_LATEST_MISSING_PARAMETER = {
-    "from_block": {"block_number": 0},
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "to_block": "latest",
-    "chunk_size": 10,
-}
-
-BLOCK_FROM_0_TO_LATEST_WRONG_BLOCK_TYPE = {
-    "from_block": {"block_number": "0x0"},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "to_block": "latest",
-    "chunk_size": 10,
-}
-
-BLOCK_FROM_0_TO_2 = {
-    "from_block": {"block_hash": rpc_felt("0x0")},
-    "to_block": {"block_number": 2},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "chunk_size": 10,
-}
+def create_get_events_filter(
+    *,
+    from_block: Union[int, str] = 0,
+    to_block: Union[int, str] = "latest",
+    address: Optional[str] = rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
+    keys: Optional[List[str]] = None,
+    chunk_size: int = 10,
+    continuation_token: Optional[str] = None
+) -> dict:
+    """
+    Method that creates ``filter`` parameter for `get_events` RPC call.
```

Comment:
This is more docstring than I would expected, thanks for that! Probably unimportant, but for it's just that some of the optional parameters indicate that they are optional while others don't.

Reply:
I think if we have a default value that is not `None` we shouldn't be using `Optional`. It looks okay to me.

Reply:
`Optional[...]` is a shorthand notation for `Union[..., None]` so it might make sense to have it like this if it can be `None` (if we are talking about `address`).
[Source](https://stackoverflow.com/a/51710151)

Reply:
Used `Optional` for address because of one fixture in one test (particularly for `test_get_events_missing_parameter`), however now I think it isn't necessary - the dictionary can be just changed after calling the function (like in test below that one - `test_get_events_wrong_blockid_type`). Tell me what you prefer.

Reply:
Modifying the dict is better IMO.